Mathlib/Data/Matroid/Constructions.lean

```diff
@@ -0,0 +1,242 @@
+    and_iff_left (inter_subset_right _ _)]
+
+theorem eq_freeOn_iff : M = freeOn E ↔ M.E = E ∧ M.Indep E := by
+  refine ⟨?_, fun h ↦ ?_⟩
+  · rintro rfl; simp [Subset.rfl]
+  simp only [eq_iff_indep_iff_indep_forall, freeOn_ground, freeOn_indep_iff, h.1, true_and]
+  exact fun I hIX ↦ iff_of_true (h.2.subset hIX) hIX
+
+theorem ground_indep_iff_eq_freeOn : M.Indep M.E ↔ M = freeOn M.E := by
+  simp [eq_freeOn_iff]
+
+theorem freeOn_restrict (h : R ⊆ E) : (freeOn E) ↾ R = freeOn R := by
+  simp [h, eq_freeOn_iff, Subset.rfl]
+
+end FreeOn
+
+section uniqueBaseOn
+
+/-- The matroid on `E` whose unique base is the subset `I` of `E`.
+Intended for use when `I ⊆ E`; if this not not the case, then the base is `I ∩ E`. -/
+def uniqueBaseOn (I E : Set α) : Matroid α := (freeOn I) ↾ E
+
+@[simp] theorem uniqueBaseOn_ground : (uniqueBaseOn I E).E = E :=
+  rfl
+
+theorem uniqueBaseOn_base_iff (hIE : I ⊆ E) : (uniqueBaseOn I E).Base B ↔ B = I := by
+  rw [uniqueBaseOn, base_restrict_iff', freeOn_basis'_iff, inter_eq_self_of_subset_right hIE]
+
+theorem uniqueBaseOn_inter_ground_eq (I E : Set α) :
+    uniqueBaseOn (I ∩ E) E = uniqueBaseOn I E := by
+  simp only [uniqueBaseOn, restrict_eq_restrict_iff, freeOn_indep_iff, subset_inter_iff,
+    iff_self_and]
+  tauto
+
+@[simp] theorem uniqueBaseOn_indep_iff' : (uniqueBaseOn I E).Indep J ↔ J ⊆ I ∩ E := by
+  rw [uniqueBaseOn, restrict_indep_iff, freeOn_indep_iff, subset_inter_iff]
+
+theorem uniqueBaseOn_indep_iff (hIE : I ⊆ E) : (uniqueBaseOn I E).Indep J ↔ J ⊆ I := by
+  rw [uniqueBaseOn, restrict_indep_iff, freeOn_indep_iff, and_iff_left_iff_imp]
+  exact fun h ↦ h.trans hIE
+
+theorem uniqueBaseOn_basis_iff (hI : I ⊆ E) (hX : X ⊆ E) :
+    (uniqueBaseOn I E).Basis J X ↔ J = X ∩ I := by
+  rw [basis_iff_mem_maximals]
+  simp_rw [uniqueBaseOn_indep_iff', ← subset_inter_iff, ← le_eq_subset, Iic_def, maximals_Iic,
+    mem_singleton_iff, inter_eq_self_of_subset_left hI, inter_comm I]
+
+theorem uniqueBaseOn_inter_basis (hI : I ⊆ E) (hX : X ⊆ E) :
+    (uniqueBaseOn I E).Basis (X ∩ I) X := by
+  rw [uniqueBaseOn_basis_iff hI hX]
+
+@[simp] theorem uniqueBaseOn_dual_eq (I E : Set α) :
+    (uniqueBaseOn I E)✶ = uniqueBaseOn (E \ I) E := by
+  rw [← uniqueBaseOn_inter_ground_eq]
+  refine eq_of_base_iff_base_forall rfl (fun B (hB : B ⊆ E) ↦ ?_)
+  rw [dual_base_iff, uniqueBaseOn_base_iff (inter_subset_right _ _),
+    uniqueBaseOn_base_iff (diff_subset _ _), uniqueBaseOn_ground]
+  refine' ⟨fun h ↦ _, fun h ↦ _⟩
+  · rw [← diff_diff_cancel_left hB, h, diff_inter_self_eq_diff]
+  rw [h, inter_comm I]; simp
+
+@[simp] theorem uniqueBaseOn_self (I : Set α) : uniqueBaseOn I I = freeOn I := by
+  rw [uniqueBaseOn, freeOn_restrict rfl.subset]
+
+@[simp] theorem uniqueBaseOn_empty (I : Set α) : uniqueBaseOn ∅ I = loopyOn I := by
+  rw [← dual_inj, uniqueBaseOn_dual_eq, diff_empty, uniqueBaseOn_self, loopyOn_dual_eq]
+
+@[simp] theorem uniqueBaseOn_restrict' (I E R : Set α) :
```

Comment:
There's a chance you don't want this one as a `@[simp]` lemma, since you'd rather use `uniqueBaseOn_restrict` if you have that hypothesis available, and I'm not sure there are any simp lemmas that say `I ∩ R ∩ E = I ∩ R` if `I ⊆ E`, so you would lose confluence by including it.